### DISCUSSION TITLE
improving docker documentation

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#167] [Design & Development]
I physically went to the office today to mess with docker on this mac pro.   I think this is a good test machine for an example desktop system a typical seismologists would want to install mspass on to get started. Some experiences:

1.  First, I am not like a typical user who would likely have never messed with docker before.   I'd followed the wiki procedure numerous times before.  Maybe that just confused me.
2.  I looked at this a week ago and was confused then about how the two different pages in the user manual documentation are or are not related.  Before I tried running them sequentially which was a real disaster.  
3.  This time I followed  the first page title 'Run MsPASS with Docker".   Worked fine to get tutorials working with Jupyter that don't need MongoDB.  Is that the idea of that page?   
4.  Sorry to say I am totally confused by the second section titled "Deploy MsPASS with Docker Compose".   Is that the procedure I should follow to get an instance of mspass running in docker that has a database and spark running? 

I think one thing I need to do is completely clear the contents of what done so far.  I have no idea how to do that.

This raises a final point I make as joe scientist who is more knowledgeable than 99% of our user's will be.  We really will have to have a basic tutorial using docker that as you work through the tutorial will introduce some of the concepts.   Docker is, from what I see as rookie at this point, a fairly complicated piece of software. The problem I have with any tutorial I find is they have all this material about setting up a container.  Our user's will not want to do that.   To make this successful we need a very clean plug it in and it will work system that people can use initially and build some confidence.  Otherwise this whole project is going to flame out.   In getting started we need a very clean set of instruction to get docker running for some stated purpose (presumably running all the tutorials we supply)     Presumably documentation on how to set this up on an hpc system will be a different challenge, but may have a smaller, more advanced audience.

**Comment by wangyinz** [comment]
Well, actually if you were able to get through "Run MsPASS with Docker", you are already 100% done with "getting started". Inside the launched notebook, users have full access to the database and scheduler and that's why it's called the "all-in-one mode". 

The "Deploy MsPASS with Docker Compose" is really an example to deploy mspass in a distributed environment, which is (at least for now) not necessarily written for those that have no knowledge of what a distributed environment means. Also, I do expect we revise that section a lot once we have it tested our on the Cloud and HPC, and we will then have a more specific instruction for the different setups. The Docker Compose example shown right now is just simulating a distributed environment with multiple docker containers.
